### PR TITLE
Feature/17630 boost stacking tooltip pointpercentage value

### LINKED ISSUE #17630: Boost & stacking: tooltip point.percentage value
In boost, the point is stripped down with most of its options, e.g. `point.percentage`: https://jsfiddle.net/BlackLabel/tkwrgL24/
image

Feature description: add `point.percentage` value to a stacked boosted point so it's available in a tooltip.

PoC: based on quite old v7.2.1, but I hope it will be similar in v10+
https://jsfiddle.net/BlackLabel/swm47ebd/
TO DO:
- add `point.percentage = boostPoint.percentage;` in the `getPoint` method
- add `percentage: percentage` in the `addKDPoint` method (and its parameters)
----

**You can vote for this feature by adding a thumbs-up reaction to this post.**

### PR DESCRIPTION
**Pull request by MarkusBarstad**
Fixed #17630, making it possible to display percentage values when hovering over a boosted stacked point.

### CODE REVIEW

File: ts/Extensions/Boost/BoostSeries.ts [hunk 1]
```diff
@@ -1015,7 +1018,7 @@ function seriesRenderCanvas(this: Series): void {
      * @private
      */
     function processPoint(
-        d: (number|Array<number>|Record<string, number>),
+        d: (number|Array<number>|Record<string, number>|number),
```
Comment (TorsteinHonsi) [inline ts/Extensions/Boost/BoostSeries.ts:1021]: Istn't the `number` type already covered in the beginning of the line?

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, and congrats on your first PR 🎉 

I think this should be considered a bugfix rather than an enhancement. Please see https://github.com/highcharts/highcharts/tree/master/changelog#writing-content-for-the-changelog for how to form the description and what tags to use.

File: ts/Extensions/Boost/BoostSeries.ts (hunk 1 above)
Comment (MarkusBarstad) [reply ts/Extensions/Boost/BoostSeries.ts:1021]: Yes, good point!

**Comment by MarkusBarstad** [comment]
Thank you 🎉 !! 

I will "reformat" this to a bugfix as you request! Do you want me to also rename the branch appropriately?

**Comment by TorsteinHonsi** [comment]
> Do you want me to also rename the branch appropriately?

No need for that, it will soon be history

**Comment by MarkusBarstad** [comment]
I edited the first comment attempting to follow the changelog-guidelines, but I am a bit unsure if this counts as its description x)

I also removed the needless or-ed `number`-type as you requested!

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**